$OpenBSD: patch-xwindow_x_draw_str_c,v 1.1 2014/04/29 11:19:50 dcoppa Exp $

x_draw_str.c: The margin area of transparent frame of animation gif
is never cleared (upstream commit d05bf410efcc7c140435c18445325254281008e4)

--- xwindow/x_draw_str.c.orig	Tue Apr 29 13:05:57 2014
+++ xwindow/x_draw_str.c	Tue Apr 29 13:07:32 2014
@@ -129,7 +129,7 @@ draw_picture(
 			}
 		}
 
-		if( need_clear)
+		if( need_clear > 0)
 		{
 			x_window_clear( window , dst_x , dst_y , dst_width , line_height) ;
 		}
@@ -175,7 +175,7 @@ draw_picture(
 		if( strstr( cur_pic->file_path , "mlterm/animx") && cur_pic->next_frame >= 0)
 		{
 			/* Don't clear if cur_pic is 2nd or later GIF Animation frame. */
-			need_clear = 0 ;
+			need_clear = -1 ;
 		}
 
 		if( (src_width = w) < ch_width && ! need_clear)
